activerecord/lib/active_record/relation/batches.rb

```diff
@@ -274,46 +304,58 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
           end
         end
 
-        batch_relation = batch_condition(
-          relation, primary_key, primary_key_offset, order == :desc ? :lt : :gt
-        )
+        batch_orders_copy = batch_orders.dup
+        _last_column, last_order = batch_orders_copy.pop.first
```

Comment:
I don't think this actually works as we don't need to call `.first` as long as we are splatting the array into two variables

```suggestion
        _last_column, last_order = batch_orders_copy.pop
```

Otherwise `last_order` will be `nil`:

```ruby
irb(main):008:0> batch_orders_copy = [[:col1, :asc], [:col2, :desc]]
=> [[:col1, :asc], [:col2, :desc]]
irb(main):009:0> _last_column, last_order = batch_orders_copy.pop.first
=> :col2
irb(main):010:0> _last_column
=> :col2
irb(main):011:0> last_order
=> nil
```

Reply:
@nvasilevski
After [this comment](https://github.com/rails/rails/pull/48268#discussion_r1202822901) I changed the return value of the build_batch_orders method to the following
```
{:col1 => :asc, :col2 => :desc}
```
 and after [this comment](https://github.com/rails/rails/pull/48268#discussion_r1204636132) I changed the return value to the following. 
```
[{:col1 => :asc}, {:col2 => :desc}]
```
But maybe the return value you intended was something like the following?
```
[[:col1, :asc], [:col2, :desc]]
```
 If so, I will modify the code!